### PR TITLE
#284 Sign in for administration frontend

### LINKED ISSUE #284: Basic login to administration BE
Usage of the administration BE should only be able after login.

_Out of scope_:
- creating new users (#285)
- rights and roles (#286)

Part of #280

### PR DESCRIPTION
**Pull request by elKei24**
Closes #284

### CODE REVIEW

**Comment by Schedulaar** [comment]
I added a default user:
`admin@example.com` with password `admin` which gets created if no user exists

**Review by Schedulaar — APPROVED** [review]

File: administration/__generated__/globalTypes.ts [hunk 1]
```diff
@@ -7,6 +7,11 @@
 // START Enums and Input Objects
 //==============================================================
 
+export interface AuthDataInput {
+  email: string;
+  password: string;
```
Comment (maxammann) [inline administration/__generated__/globalTypes.ts:12]: Should this be a passwordHash?

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/common/webservice/WebService.kt [hunk 2]
```diff
@@ -22,6 +22,10 @@ class WebService {
         println("Goto http://${host}:${port}/graphiql for a graphical editor")
 
         val graphQLHandler = GraphQLHandler()
-        app.post("/") { ctx -> graphQLHandler.handle(ctx) }
+        app.post("/") { ctx ->
+            ctx.header("Access-Control-Allow-Headers: Authorization")
+            ctx.header("Access-Control-Allow-Origin: *")
```
Comment (maxammann) [inline backend/src/main/kotlin/app/ehrenamtskarte/backend/common/webservice/WebService.kt:27]: This should only be done in development and not production.

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/auth/database/Setup.kt [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
 
+import app.ehrenamtskarte.backend.auth.database.repos.AdministratorsRepository
 import org.jetbrains.exposed.sql.SchemaUtils
 
 fun setupDatabase() {
     SchemaUtils.create(
         Administrators
     )
+    if (AdministratorEntity.count() == 0L) {
+        AdministratorsRepository.insert("admin@example.com", "admin")
```
Comment (maxammann) [inline backend/src/main/kotlin/app/ehrenamtskarte/backend/auth/database/Setup.kt:11]: `admin` is the PW here, right? I think we should use bcrypt here to store the passwords.

Comment (maxammann) [reply backend/src/main/kotlin/app/ehrenamtskarte/backend/auth/database/Setup.kt:11]: Then we also can set a default password by just using the hashed version of it.

File: administration/__generated__/globalTypes.ts (hunk 1 above)
Comment (maxammann) [reply administration/__generated__/globalTypes.ts:12]: Else we are transmitting plain text passwords to the server.

**Review by maxammann — APPROVED** [review]
I think we should move the issues I mentioned to a separate issue as it is probably quite huge.

File: administration/__generated__/globalTypes.ts (hunk 1 above)
Comment (Schedulaar) [reply administration/__generated__/globalTypes.ts:12]: I created an issue. https://github.com/ehrenamtskarte/ehrenamtskarte/issues/322

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/auth/database/Setup.kt (hunk 3 above)
Comment (Schedulaar) [reply backend/src/main/kotlin/app/ehrenamtskarte/backend/auth/database/Setup.kt:11]: `admin` is the pw, yes. `AdminstratorsRepository` does use bcrypt under the hood though...

**Pull request merged**